cpp/include/patterns/copy_v_transform_reduce_nbr.cuh

```diff
@@ -0,0 +1,293 @@
+        init);
+  }
+}
+
+/**
+ * @brief Iterate over the outgoing edges to update vertex properties.
+ *
+ * This function is inspired by thrust::transfrom_reduce() (iteration over the outgoing edges part)
+ * and thrust::copy() (update vertex properties part, take transform_reduce output as copy input).
+ *
+ * @tparam GraphViewType Type of the passed non-owning graph object.
+ * @tparam AdjMatrixRowValueInputIterator Type of the iterator for graph adjacency matrix row
+ * input properties.
+ * @tparam AdjMatrixColValueInputIterator Type of the iterator for graph adjacency matrix column
+ * input properties.
+ * @tparam EdgeOp Type of the binary (or ternary) edge operator.
+ * @tparam T Type of the initial value for reduction over the outgoing edges.
+ * @tparam VertexValueOutputIterator Type of the iterator for vertex output property variables.
+ * @param handle RAFT handle object to encapsulate resources (e.g. CUDA stream, communicator, and
+ * handles to various CUDA libraries) to run graph algorithms.
+ * @param graph_view Non-owning graph object.
+ * @param adj_matrix_row_value_input_first Iterator pointing to the adjacency matrix row input
+ * properties for the first (inclusive) row (assigned to this process in multi-GPU).
+ * `adj_matrix_row_value_input_last` (exclusive) is deduced as @p adj_matrix_row_value_input_first +
+ * @p graph_view.get_number_of_adj_matrix_local_rows().
+ * @param adj_matrix_col_value_input_first Iterator pointing to the adjacency matrix column input
+ * properties for the first (inclusive) column (assigned to this process in multi-GPU).
+ * `adj_matrix_col_value_output_last` (exclusive) is deduced as @p adj_matrix_col_value_output_first
+ * + @p graph_view.get_number_of_adj_matrix_local_cols().
+ * @param e_op Binary (or ternary) operator takes *(@p adj_matrix_row_value_input_first + i), *(@p
+ * adj_matrix_col_value_input_first + j), (and optionally edge weight) (where i and j are row and
+ * column indices, respectively) and returns a value to be reduced.
+ * @param init Initial value to be added to the reduced @e_op return values for each vertex.
+ * @param vertex_value_output_first Iterator pointing to the vertex property variables for the first
+ * (inclusive) vertex (assigned to tihs process in multi-GPU). `vertex_value_output_last`
+ * (exclusive) is deduced as @p vertex_value_output_first + @p
+ * graph_view.get_number_of_local_vertices().
+ */
+template <typename GraphViewType,
+          typename AdjMatrixRowValueInputIterator,
+          typename AdjMatrixColValueInputIterator,
+          typename EdgeOp,
+          typename T,
+          typename VertexValueOutputIterator>
+void copy_v_transform_reduce_out_nbr(
+  raft::handle_t const& handle,
+  GraphViewType const& graph_view,
+  AdjMatrixRowValueInputIterator adj_matrix_row_value_input_first,
+  AdjMatrixColValueInputIterator adj_matrix_col_value_input_first,
+  EdgeOp e_op,
+  T init,
+  VertexValueOutputIterator vertex_value_output_first)
```

Comment:
Suggest moving the output iterator to be the fifth parameter (before `EdgeOp e_op`). This would be more consistent with `thrust::transform` and `thrust::copy_if`.

Reply:
See the reply to "copy_v_transform_reduce_in_nbr". Same here.